spek-samples/src/test/kotlin/org/jetbrains/spek/samples/subject/CalculatorTableSpec.kt

```diff
@@ -0,0 +1,38 @@
+package org.jetbrains.spek.samples.subject
+
+import org.jetbrains.samples.Calculator
+import org.jetbrains.spek.api.Spek
+import org.jetbrains.spek.api.dsl.describe
+import org.jetbrains.spek.api.dsl.it
+import org.jetbrains.spek.api.dsl.on
+import org.jetbrains.spek.table.example
+import org.jetbrains.spek.table.with
+import kotlin.test.assertEquals
+
+object CalculatorTableSpec : Spek({
+    describe("subtract") {
+        with(
+            example(4, 2),
```

Comment:
Just use 3 params and assume the 3rd parameter is the expected value `example(a, b, result)`.  In some cases you don't have an expected value to assert but an expected side-effect (exception thrown, etc ...) when certain parameters are used.

```kotlin
forEvery(
    ...
) {
    expect({ some.foo(param1, param2) }).throws<SomeException>()
}
```